.github/workflows/docs.yml

```diff
@@ -17,28 +17,36 @@ jobs:
     name: Publish
     runs-on: ubuntu-latest
     steps:
-      - name: Apt Dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y build-essential clang ocl-icd-opencl-dev
       - name: Checkout Sources
         uses: actions/checkout@v3
+      - name: Apt Dependencies
+        run: sudo make install-deps
       - name: Link Checker (Repo)
         uses: lycheeverse/lychee-action@v1.5.1
         with:
           args: --verbose --no-progress 'README.md'
           # Fail action on broken links
           fail: true
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: 5
-        continue-on-error: true
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+          key: ${{ runner.os }}-docs-cargo-bin-${{ hashFiles('rust-toolchain.toml') }}
       - name: Install MDBook
         run: cargo install mdbook mdbook-linkcheck
+        continue-on-error: true
+      - name: Setup sccache
+        uses: hanabi1224/sccache-action@v1.2.0
```

Comment:
it would be great to include a comment on why are we using a fork

Reply:
Done. Basically we need the `cache-update` to only save the cache on 1. `main` branch, 2. cache not found. The original action repo looks like unmaintained so we keep using the fork